### PR TITLE
Cubemap update

### PR DESCRIPTION
**Pull request by slimbuck**
This PR makes the following changes:
- support loading cubemap without prefiltered lighting data
- don't fire events on cubemap patch (as the old cubemap handler did). instead rely on asset:change events.
- only set cubemap resources if they've changed
- don't load helipad faces in the viewer (which aren't in github).
- fix the engine bundle tests

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: tools/viewer/src/viewer.js [hunk 1]
```diff
@@ -340,42 +340,39 @@ Object.assign(Viewer.prototype, {
...
-                assetsFolder + "/cubemaps/Helipad_negz.png"
-            ],
+            // textures: [
+            //     assetsFolder + "/cubemaps/Helipad_posx.png",
+            //     assetsFolder + "/cubemaps/Helipad_negx.png",
+            //     assetsFolder + "/cubemaps/Helipad_posy.png",
+            //     assetsFolder + "/cubemaps/Helipad_negy.png",
+            //     assetsFolder + "/cubemaps/Helipad_posz.png",
+            //     assetsFolder + "/cubemaps/Helipad_negz.png"
+            // ],
```
Comment (willeastcott) [inline tools/viewer/src/viewer.js:350]: Maybe just delete rather than comment out?

File: tools/viewer/src/viewer.js [hunk 2]
```diff
@@ -340,42 +340,39 @@ Object.assign(Viewer.prototype, {
...
             magFilter: pc.FILTER_LINEAR,
             minFilter: pc.FILTER_LINEAR_MIPMAP_LINEAR,
             anisotropy: 1,
             type: pc.TEXTURETYPE_RGBM
         });
-        cubemap.loadFaces = true;
+        // cubemap.loadFaces = true;
```
Comment (willeastcott) [inline tools/viewer/src/viewer.js:356]: Delete?

File: src/resources/cubemap.js [hunk 3]
```diff
@@ -35,13 +35,13 @@ Object.assign(CubemapHandler.prototype, {
...
+                // fire error event if patch failed
                 registry.fire('error', asset);
                 registry.fire('error:' + asset.id, err, asset);
                 asset.fire('error', asset);
             } else {
-                registry.fire('load', asset);
-                registry.fire('load:' + asset.id, asset);
-                asset.fire('load', asset);
+                // nothing to do since asset:change would have been raised if
+                // resources were changed.
```
Comment (willeastcott) [inline src/resources/cubemap.js:44]: This looks odd. Maybe just `// no else clause required because...`?

File: tools/viewer/src/viewer.js (hunk 2 above)
Comment (slimbuck) [reply tools/viewer/src/viewer.js]: I left these as a reminder to myself and others what's needed to load faces. I've removed though.

File: tools/viewer/src/viewer.js (hunk 1 above)
Comment (slimbuck) [reply tools/viewer/src/viewer.js]: removed

File: src/resources/cubemap.js (hunk 3 above)
Comment (slimbuck) [reply src/resources/cubemap.js]: done

**Pull request merged**